homeassistant/components/vivotek/camera.py

```diff
@@ -0,0 +1,123 @@
+"""Support for Vivotek IP Cameras."""
+
+import logging
+
+import voluptuous as vol
+from libpyvivotek import VivotekCamera
+
+from homeassistant.const import (
+    CONF_IP_ADDRESS,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_SSL,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
+from homeassistant.helpers import config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_FRAMERATE = "framerate"
+
+DEFAULT_NAME = "Vivotek Camera"
+DEFAULT_EVENT_0_KEY = "event_i0_enable"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_IP_ADDRESS): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_SSL, default=False): cv.boolean,
+        vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
+        vol.Optional(CONF_FRAMERATE, default=2): cv.positive_int,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a Vivotek IP Camera."""
+    add_entities([VivotekCam(config)])
+
+
+class VivotekCam(Camera):
+    """A Vivotek IP camera."""
+
+    def __init__(self, config):
+        """Initialize a Vivotek camera."""
+        super().__init__()
+
+        self._frame_interval = 1 / config[CONF_FRAMERATE]
+        self._name = config[CONF_NAME]
+        self._motion_detection_enabled = False
+        self._event_0_key = DEFAULT_EVENT_0_KEY
```

Comment:
We don't need an instance attribute for the event key. We can use the constant `DEFAULT_EVENT_0_KEY` directly.